instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/engine/unit/UnitConverterFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jmx.engine.unit;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+public class UnitConverterFactory {
+
+  private static final Map<String, Map<String, UnitConverter>> conversionMappings = new HashMap<>();
```

Comment:
One way to get rid of the map would be to create a composite key that combines both source and destination metrics. For example `ms_s` as key to convert `ms` to `s`, this is doable because the map is an implementation detail and is not exposed.